src/components/ha-blueprint-picker.ts

```diff
@@ -0,0 +1,124 @@
+import "@polymer/paper-dropdown-menu/paper-dropdown-menu-light";
+import "@polymer/paper-item/paper-item";
+import "@polymer/paper-listbox/paper-listbox";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../common/dom/fire_event";
+import { compare } from "../common/string/compare";
+import { Blueprints, fetchBlueprints } from "../data/blueprint";
+import { HomeAssistant } from "../types";
+
+@customElement("ha-blueprint-picker")
+class HaBluePrintPicker extends LitElement {
+  public hass?: HomeAssistant;
+
+  @property() public label?: string;
+
+  @property() public value = "";
+
+  @property() public domain = "automation";
+
+  @property() public blueprints?: Blueprints;
+
+  @property({ type: Boolean }) public disabled = false;
+
+  private _sortedBlueprints = memoizeOne((blueprints?: Blueprints) => {
+    if (!blueprints) {
+      return [];
+    }
+    let result: any[] = [];
+    result = result.concat(
+      Object.entries(blueprints).map(([path, blueprint]) => ({
+        ...blueprint.metadata,
+        path,
+      }))
+    );
```

Comment:
Why concat on an empty list? 
```suggestion
    const result = Object.entries(blueprints).map(([path, blueprint]) => ({
      ...blueprint.metadata,
      path,
    }));
```